<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{XBSeq for testing differential expression for RNA-seq}
-->
# Introduction 

XBSeq is a novel algorithm for testing RNA-seq differential expression (DE), where a statistical model was established
based on the assumption that observed signals are the convolution of true expression signals and sequencing noises. The
mapped reads in non-exonic regions are considered as sequencing noises, which follows a Poisson distribution. Given
measurable observed and noise signals from RNA-seq data, true expression signals, assuming governed by the negative
binomial distribution, can be delineated and thus the accurate detection of differential expressed genes

# Installation 

XBSeq can be installed from Bioconductor by 
```r
source('http://www.bioconductor.org/biocLite.R')
biocLite("XBSeq")
```
```{r}
library("XBSeq")
```
If you would like to install the development version of XBSeq, it is recommended 
that you refer to the github page of XBSeq. 

XBSeq imports DESeq2, Biobase, locfit, pracma, matrixStats, ggplot2, etc. from Bioconductor

# Use XBSeq for testing differential expression 

### HTseq counting

In order to use XBSeq for testing DE, after sequence alignment, we need to run HTSeq twice to measure the reads mapped to exonic regions (observed signal) and non-exonic regions (background noise). Generally speaking, you will need to run the following code to generate observed read count and background read count. 

```
htseq-count [options] <alignment_file> <gtf_file> > Observed_count.txt
htseq-count [options] <alignment_file> <gtf_file_bg> > background_count.txt
```

Details regarding how HTSeq works can be found here:
http://www-huber.embl.de/HTSeq/doc/count.html

The gtf file used to measure observed signal can be downloaded from UCSC database: http://genome.ucsc.edu. The gtf file used to measure background noise can be downloaded in the gtf folder from github: https://github.com/Liuy12/XBSeq. Details regarding the procedure we used to construct the background gtf file can be found in the Details section 

### XBSeq testing for DE 

After HTSeq procedure, then we will have two measurements for each gene, the observed signal and background noise. Here we will use a mouse RNA-seq dataset, which contains 3 replicates of wild type mouse liver tissues (WT) and 3 replicates of Myc transgenic mouse liver tissues (MYC). The dataset is obtained from Gene Expression Omnibus [(GSE61875)](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE61875). 

As a preliminary step, we have already carried out HTSeq procedure mentioned above to generate observed signal and background noise for each gene. The two datasets can be loaded into user's working space by 

```{r}
data(ExampleData)
```

We can first take a look at the two datasets:

```{r}
head(Observed)
head(Background)
```

Rows represent reads mapped to each gene or corresponding background region. Column represent samples. And differential expression analysis will be carried out as follows:

Observe and background are the output matrix from HTSeq

```{r}
Signal <- estimateRealcount(Observed, Background)
```

Conditions are the design matrix for the experiment

```{r}
conditions <- factor(c(rep('C',3), rep('T', 3)))
XB <- XBSeqDataSet(Signal,conditions)
XB <- estimateSizeFactors( XB )
XB <-estimateSCV( XB, Observed, Background, method='pooled', sharingMode='maximum', fitType='local' )
```

Take a look at the scv fitting information 

```{r}
plotSCVEsts(XB)
```

Carry out the DE test by using function XBSeqTest

```{r}
Teststas <- XBSeqTest( XB, levels(conditions)[1L], levels(conditions)[2L] )
```

Plot Maplot based on test statistics

```{r}
MAplot(Teststas, padj = FALSE, pcuff = 0.01, lfccuff = 1)
```

```r
# Alternatively, all the codes above can be done with a wrapper function XBSeq
Teststats <- XBSeq( Observed, Background, conditions, method='pooled', sharingMode='maximum', fitType='local', pvals_only=FALSE )
```

### Compare the results with DESeq

Now we will carry out DE analysis on the same dataset by using DESeq and then compare the results obtained by these two methods

If you have not installed DESeq before, DESeq is also available from Bioconductor

```r
biocLite("DESeq")
```

Then DE analysis for DESeq can be carried out by:

```{r}
library('DESeq')
library('ggplot2')
de <- newCountDataSet(Observed, conditions)
de <- estimateSizeFactors(de)
de <- estimateDispersions(de, method = "pooled", fitType="local")
res <- nbinomTest(de, levels(conditions)[1], levels(conditions)[2])
```

Then we can compare the results from XBSeq and DESeq

```{r}
DE_index_DESeq <- with(res, which(pval<0.01 & abs(log2FoldChange)>1))
DE_index_XBSeq <- with(Teststas, which(pval<0.01 & abs(log2FoldChange)>1))
DE_index_inters <- intersect(DE_index_DESeq, DE_index_XBSeq)
DE_index_DESeq_uniq <- setdiff(DE_index_DESeq, DE_index_XBSeq)
DE_plot <- MAplot(Teststas, padj = FALSE, pcuff = 0.01, lfccuff = 1, shape=16)
DE_plot + geom_point( data=Teststas[DE_index_inters,], aes( x=baseMean, y=log2FoldChange ), color= 'green', shape=16 ) + geom_point( data=Teststas[DE_index_DESeq_uniq,], aes( x=baseMean, y=log2FoldChange ), color= 'blue', shape=16 )
```

The red dots indicate DE genes identified only by XBSeq. Then green dots are the shared results of XBSeq and DESeq. The blue dots are DE genes identified only by DESeq. 

# Details 

### Construction of gtf file for background region

* Download refFlat table from UCSC database (http://genome.ucsc.edu) and create the preliminary list of gene-free regions,

* Download tables of (a) all_mrna; (b) ensGene; (c) pseudoYale60Gene; (d) vegaGene;, (e)xenoMrna, and (f) xenoRefGene from UCSC database and remove regions appear in any of them from the gene-free regions,

* To guarantee gene-free regions are far enough from exonic regions, trim 100 bps from both sides of intronic regions and 1,000 bps from both sides of inter-genic regions,

* Shift each exon of a gene to the right nearest gene-free region.  Most of the shifted genes remain the same as the original structures of the genes,

* If the nearby gene-free region is too short, we may only preserve the exon size features but not the whole gene structure. The priority of shifting a region is: i) nearest right gene-free region, 2) nearest left gene-free region; 3) the second right nearest gene-free region and so on until the shift region of the original exon fits, and

* At last, we considered the shifted regions as the non-exonic regions for each gene and a final .gtf file was created. 

# Bug reports
Report bugs as issues on our [GitHub repository](https://github.com/Liuy12/XBSeq/issues) or you can report directly to my email: liuy12@uthscsa.edu.

# Session information 
```{r}
sessionInfo()
```

# Acknowledgements 
XBSeq is implemented in R based on the source code from DESeq and DESeq2. 

# References
Hung-I Harry Chen, Yuanhang Liu, Yi Zou, Zhao Lai, Devanand Sarkar, Yufei Huang, Yidong Chen, doi: http://dx.doi.org/10.1101/016196